src/v/security/authorizer.h

```diff
@@ -211,20 +211,29 @@ class authorizer final {
         auto acls = _store.find(type, resource_name());
 
         if (_superusers.contains(principal)) {
-            return true;
+            return auth_result::superuser_authorized(
+              principal, host, resource_name);
         }
 
         if (acls.empty()) {
-            return bool(_allow_empty_matches);
+            return auth_result::empty_match_result(
+              principal, host, resource_name, bool(_allow_empty_matches));
         }
 
         // check for deny
-        if (acls.contains(operation, principal, host, acl_permission::deny)) {
-            return false;
+        if (auto entry = acls.contains(
+              operation, principal, host, acl_permission::deny);
+            entry.has_value()) {
```

Comment:
does it seem odd to have a method called `.contains()` return anything other than a bool? here it seems to return something that has a `has_value` method.

Reply:
yup, point taken